src/ctelnet.cpp

```diff
@@ -1202,11 +1202,15 @@ void cTelnet::processTelnetCommand(const std::string& command)
                     hisOptionState[idxOption] = true;
                 } else if ((option == OPT_COMPRESS) || (option == OPT_COMPRESS2)) {
                     //these are handled separately, as they're a bit special
-                    if (mpHost->mFORCE_NO_COMPRESSION || ((option == OPT_COMPRESS) && (hisOptionState[static_cast<int>(OPT_COMPRESS2)]))) {
+                    if (mpHost->mFORCE_NO_COMPRESSION) {
+                        sendTelnetOption(TN_DONT, option);
+                        hisOptionState[idxOption] = false;
+                        qDebug().nospace().noquote() << "Rejecting MCCP v" << (option == OPT_COMPRESS ? "1" : "2") << ", because FORCE COMPRESSION OFF is set to ON.";
```

Comment:
```suggestion
                        qDebug().nospace().noquote() << "Rejecting MCCP v" << (option == OPT_COMPRESS ? "1" : "2") << ", because FORCE COMPRESSION OFF is enabled.";
```

Reply:
🤔 So that the setting is a bit clearer let us quote the original (Engineering English) option literally:
```suggestion
                        qDebug().nospace().noquote() << "Rejecting MCCP v" << (option == OPT_COMPRESS ? "1" : "2") << ", because the 'Force compression off' option is enabled.";
```